modules/fedimint-lnv2-server/src/lib.rs

```diff
@@ -0,0 +1,668 @@
+                            sk: sks[peer.to_usize()],
+                        },
+                    }
+                    .to_erased(),
+                )
+            })
+            .collect();
+
+        server_cfg
+    }
+
+    async fn distributed_gen(
+        &self,
+        peers: &PeerHandle,
+        params: &ConfigGenModuleParams,
+    ) -> DkgResult<ServerModuleConfig> {
+        let params = self.parse_params(params).unwrap();
+        let g1 = peers.run_dkg_g1(()).await?;
+        let (poly_g1, sk) = g1[&()].clone().tpe();
+
+        let server = LightningConfig {
+            local: LightningConfigLocal {
+                bitcoin_rpc: params.local.bitcoin_rpc.clone(),
+            },
+            consensus: LightningConfigConsensus {
+                tpe_agg_pk: tpe::AggregatePublicKey(poly_g1[0].to_affine()),
+                tpe_pks: peers
+                    .peer_ids()
+                    .iter()
+                    .map(|peer| {
+                        let pk = evaluate_polynomial_g1(
+                            &poly_g1,
+                            &Scalar::from(peer.to_usize() as u64 + 1),
+                        );
+
+                        (*peer, PublicKeyShare(pk))
+                    })
+                    .collect(),
+                fee_consensus: Default::default(),
+                network: params.consensus.network,
+            },
+            private: LightningConfigPrivate {
+                sk: SecretKeyShare(sk),
+            },
+        };
+
+        Ok(server.to_erased())
+    }
+
+    fn validate_config(
+        &self,
+        _identity: &PeerId,
+        _config: ServerModuleConfig,
+    ) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    fn get_client_config(
+        &self,
+        config: &ServerModuleConsensusConfig,
+    ) -> anyhow::Result<LightningClientConfig> {
+        let config = LightningConfigConsensus::from_erased(config)?;
+        Ok(LightningClientConfig {
+            tpe_agg_pk: config.tpe_agg_pk,
+            tpe_pks: config.tpe_pks,
+            fee_consensus: config.fee_consensus,
+            network: config.network,
+        })
+    }
+}
+
+fn dealer_keygen(
+    threshold: usize,
+    keys: usize,
+) -> (AggregatePublicKey, Vec<PublicKeyShare>, Vec<SecretKeyShare>) {
+    let mut rng = OsRng; // FIXME: pass rng
+    let poly: Vec<Scalar> = (0..threshold).map(|_| Scalar::random(&mut rng)).collect();
+
+    let apk = (G1Projective::generator() * eval_polynomial(&poly, &Scalar::zero())).to_affine();
+
+    let sks: Vec<SecretKeyShare> = (0..keys)
+        .map(|idx| SecretKeyShare(eval_polynomial(&poly, &Scalar::from(idx as u64 + 1))))
+        .collect();
+
+    let pks = sks
+        .iter()
+        .map(|sk| PublicKeyShare((G1Projective::generator() * sk.0).to_affine()))
+        .collect();
+
+    (AggregatePublicKey(apk), pks, sks)
+}
+
+fn eval_polynomial(coefficients: &[Scalar], x: &Scalar) -> Scalar {
```

Comment:
Do we want this and `dealer_keygen` to be in the lightning server module crate? Maybe the `tpe` crate makes more sense?